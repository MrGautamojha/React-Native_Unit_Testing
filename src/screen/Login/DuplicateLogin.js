
import React from 'react'
import { Text, View } from 'react-native'
import { useSelector } from 'react-redux';
import { useState } from 'react';

export default function DuplicateLogin() {
    const { data, setData } = useState('')
    // const reduxData = useSelector((state) => state)
    // console.log("reduxData", reduxData);
    return (
        <View style={{ flex: 1, backgroundColor: '#ffff' }}>
            <View style={{ height: 170, width: 170, marginVertical: 20, marginHorizontal: 10, backgroundColor: 'red' }}>
                <Text>{"hi"}</Text>

            </View>


        </View>
    )
}
