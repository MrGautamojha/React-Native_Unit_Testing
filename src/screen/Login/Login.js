import React, { useState } from "react";
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  TouchableOpacity,
} from "react-native";
import { connect } from "react-redux";
import { navigateToClearStack } from "../../services/navigationService";

class Login extends React.PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: '',

    }

  }
  navigateToClearStack1 = () => {
    navigateToClearStack('Detail')
  }
  render() {
    return (
      <View style={styles.container} >
        <View style={{ marginTop: 60 }}></View>

        <View style={styles.inputView}>
          <TextInput
            style={styles.textInput}
            placeholder="Email"
            placeholderTextColor="#003f5c"
            value={this.state.email}
            onChangeText={(email) => {
              // setUser({ ...user, email: email })x/
              this.setState({ email: email })
            }}
            testID="EmailIds"
          />
        </View>

        <View style={styles.inputView}>
          <TextInput
            style={styles.TextInput}
            placeholder="Password"
            placeholderTextColor="#003f5c"
            secureTextEntry={true}
            value={''}
            onChangeText={(password) =>
              // setPassword(password)
              this.setState({ password: password })
            }

            testID="PasswordID"
          />
        </View>

        <TouchableOpacity>
          <Text style={styles.forgot_button}>Forgot Password?</Text>
        </TouchableOpacity>

        <TouchableOpacity style={styles.loginBtn} onPress={navigateToClearStack} testID={"NavigationID"}>
          <Text style={styles.loginText}>LOGIN</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
const mapStateToProps = function (state) {
  return {
    profile: state,
  }
}
export default connect(mapStateToProps)(Login)


const styles = StyleSheet.create({
  container: {
    flex: 1,
    // backgroundColor: 'red',
    // alignSelf: "center",
    //justifyContent: "center",
  },

  image: {
    marginBottom: 40,
  },

  inputView: {
    backgroundColor: "#FFC0CB",
    borderRadius: 30,
    width: "100%",
    height: 50,
    alignItems: "center",
    marginVertical: 20,
  },

  textInput: {
    height: 120,
    flex: 1,
    padding: 10,
    marginLeft: 20,
  },

  forgot_button: {
    height: 30,
    marginBottom: 30,
    marginHorizontal: 20
  },

  loginBtn: {
    width: "80%",
    borderRadius: 25,
    height: 50,
    alignItems: "center",
    justifyContent: "center",
    marginTop: 40,
    backgroundColor: "#FF1493",
    alignSelf: 'center'


  },
});