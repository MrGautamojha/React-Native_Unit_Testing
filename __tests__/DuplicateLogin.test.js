import React from 'react';
import Login from '../src/screen/Login/Login';
import { shallow } from 'enzyme';
import store from '../src/redux/store';
import DuplicateLogin from '../src/screen/Login/DuplicateLogin';

let wrapper, instance;
let state = {
    data: ''
}

describe('App', () => {
    beforeEach(() => {
        wrapper = shallow(<DuplicateLogin state={state} />);
        console.log(wrapper)
    })
    test('snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });
})