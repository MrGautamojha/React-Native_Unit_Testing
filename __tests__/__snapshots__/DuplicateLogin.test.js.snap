// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "height": 170,
        "marginHorizontal": 10,
        "marginVertical": 20,
        "width": 170,
      }
    }
  >
    <Text>
      hi
    </Text>
  </View>
</View>
`;
